### PR TITLE
Bug 1652681 - Auto zoom when opening graphs view

### PR DESCRIPTION
**Pull request by alexandru-io**
The y domain of the graph is shown from 0 when opening a graph using "graph" button from an alert item, unlike the top graph which shows y depending on the values of the test shown.

### CODE REVIEW

File: ui/perfherder/graphs/GraphsContainer.jsx [hunk 1]
```diff
@@ -109,9 +109,9 @@ class GraphsContainer extends React.Component {
       scatterPlotData,
     });
 
-    if (!visibilityChanged) {
```
Comment (alexandru-io) [inline ui/perfherder/graphs/GraphsContainer.jsx:112]: @sarah-clements I can see `visibilityChanged` flag in several files, and seems like a solid logic behind. I tried to change the logic to avoid zooming out to zoomDomain limits on `componentDidMount`, after adding/removing a `legendCard` and after creating an alert from tooltip, but I end up by just removing the flag to get the desired behavior. I ran `yarn test` and nothing seemed to be breaking.
Do you remember what was the logic behind `visibilityChanged`? I can guess, but I can't catch it entirely.
Thanks!

Comment (sarah-clements) [reply ui/perfherder/graphs/GraphsContainer.jsx:112]: It's used to reset the `zoom` query parameter if you add (or remove) another test to the graph. Since it's reset on the graph level we don't want the zoom param to show a mismatch between what's happening on the graph. Maybe it'd be worth adding a comment here to explain that or even changing the variable name so it's clearer.

Comment (alexandru-io) [reply ui/perfherder/graphs/GraphsContainer.jsx:112]: There are 3 places where the `zoom` is reset:
1. when adding/removing a legend card
2. when creating an alert from the tooltip
3. when opening the graph directly from the alert item (the scale from the upper graph - which depends on the datapoints - doesn't the one from the lower graph which is always shown from zero).

1. We (the sheriffs) thought that would be useful when adding/removing a legend card to not reset the zoom. Technically it is possible, but from what you say it might cause some inconsistencies. Is that right?
2. As far as I see, the datapoint are not affected at all by the creation of an alert from tooltip, so I assume this won't be an issue (?)
3. also, we are thinking that when opening the graph from alert item graph link, we could have the scales from the 2 graphs match - like zooming the lower graph to the scale from the upper one. I found a formula, not an exact one, but close enough, that uses the variation interval of the datapoints to leave space upper and lower just as the graph from top.

How does this sound?

Comment (sarah-clements) [reply ui/perfherder/graphs/GraphsContainer.jsx:112]: > 1. We (the sheriffs) thought that would be useful when adding/removing a legend card to not reset the zoom. Technically it is possible, but from what you say it might cause some inconsistencies. Is that right?
> 

If you want to do that, I don't have a problem with it. I was only answering your question about what the `visibilityChanged` logic was for. Feel free to remove it if it's no longer relevant. :)

>3. also, we are thinking that when opening the graph from alert item graph link, we could have the scales from the 2 graphs match - like zooming the lower graph to the scale from the upper one. I found a formula, not an exact one, but close enough, that uses the variation interval of the datapoints to leave space upper and lower just as the graph from top.
> 

Sure.

**Comment by ionutgoldan** [comment]
broken-graphs-view

In this screenshot, for example, the X axis legend is located in the middle of the big graph.
Also the upper graph also looks broken. It resembles a mostly horizontal line, when in fact it has some obvious perf changes.

**Comment by alexandru-io** [comment]
Alert 26822
Screenshot from 2020-09-02 14-12-20

Alert 26813
Screenshot from 2020-09-02 14-13-00

Alert 26807
Screenshot from 2020-09-02 14-13-25

@ionutgoldan please verify a test that has all-zero datapoints. That is an edge case that I couldn't check locally as I don't have that kind of alert.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]
There still seem to be problems  with the way the graphs are plotted.

**Comment by ionutgoldan** [comment]
Following [this local link](http://localhost:5000/perf.html?#/graphs?timerange=7776000&series=autoland,2144319,1,10), I noticed the following.

weird-y-axis

vs the original

original-y-axis

The Y axis presents some negative limits, which don't make sense.

**Comment by ionutgoldan** [comment]
On [another local link](http://localhost:5000/perf.html?#/graphs?highlightAlerts=1&series=autoland,2134435,1,10&timerange=7776000&zoom=1592381680000,1600151759507,161.45499999999993,873.2550000000001), I noticed the new Y axis is bit more zoomed out, for no apparent reason.

bigger-y-axis-limits

vs the original

original-y-axis-limits

**Comment by ionutgoldan** [comment]
Oh, by bad. Should have rebased on latest master.

**Review by ionutgoldan — APPROVED** [review]
Nice: it behaves as intended!

**Comment by ionutgoldan** [comment]
@sarah-clements could you double check the source code?

**Review by sarah-clements — APPROVED** [review]
Looks ok to me.

**Comment by ionutgoldan** [comment]
Found another weird bug on http://localhost:5000/perf.html#/graphs?highlightAlerts=1&series=autoland,2690898,1,1&series=mozilla-central,1935380,1,1&timerange=2592000&zoom=1597753988000,1600348804062,0,4421.570929720516

problematic-x-axis

Seems like the graph is messed up for distinct series.

**Review by ionutgoldan — CHANGES_REQUESTED** [review]
Please prepare an elaborate list of manual test cases to cover, before submitting any further review requests.
These should provide a good aid in finding the bugs in the existing implementation.

**Review by ionutgoldan — APPROVED** [review]
